File: BOJ/Python/그리디 알고리즘/2839.py
N = int(input())
arr = list()

for i in range((N//5)+1):
    if (N-5*i) % 3 == 0:
        arr.append(i+(N-5*i) // 3)

if len(arr) == 0:
    print(-1)
else:
    print(min(arr))

File: BOJ/Kotlin/그리디 알고리즘/2839.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val N = br.readLine().toInt()
    val arr = ArrayList<Int>()

    for (i in 0..N/5){
        if ((N-i*5) % 3 == 0){
            arr.add(i+(N-i*5) / 3)
        }
    }

    if (arr.size == 0){
        println(-1)
    }
    else{
        println(arr.min())
    }
}